frontend/packages/ux-editor/src/components/config/editModal/EditComponentId.tsx

```diff
@@ -16,45 +14,39 @@ export const EditComponentId = ({
   handleComponentUpdate,
   helpText,
 }: IEditComponentId) => {
-  const { components, containers } = useSelectedFormLayout();
-  const { t } = useTranslation();
-
   const handleIdChange = (id: string) => {
     handleComponentUpdate({
       ...component,
       id,
     });
   };
 
+  // TODO
+  // Missing:
+  // - helpText
+  // - Validation/Error handling
+  // - onBlur improvement
+  // - Move "ID: " to text resource
+  // - Add Test for all components
+
   return (
-    <FormField
-      id={component.id}
-      label={t('ux_editor.modal_properties_component_change_id')}
-      value={component.id}
-      onChange={handleIdChange}
-      propertyPath='definitions/component/properties/id'
-      componentType={component.type}
-      helpText={helpText}
-      customValidationRules={(value: string) => {
-        if (value !== component.id && idExists(value, components, containers)) {
-          return 'unique';
-        }
-      }}
-      customValidationMessages={(errorCode: string) => {
-        if (errorCode === 'unique') {
-          return t('ux_editor.modal_properties_component_id_not_unique_error');
-        }
-        if (errorCode === 'pattern') {
-          return t('ux_editor.modal_properties_component_id_not_valid');
-        }
-      }}
-      renderField={({ fieldProps }) => (
-        <Textfield
-          {...fieldProps}
-          name={`component-id-input${component.id}`}
-          onChange={(e) => fieldProps.onChange(e.target.value, e)}
-        />
-      )}
-    />
+    <div>
+      <StudioTextField
+        viewProps={{
+          children: `ID: ${component.id}`,
+          variant: 'tertiary',
+          fullWidth: true,
+          style: { paddingLeft: 0, paddingRight: 0 },
+        }}
+        inputProps={{
+          prefix: <KeyVerticalIcon className={classes.prefixKeyIcon} />,
```

Comment:
The prefix prop only accepts a string, so setting the icon here results in an error on build. This icon should be moved to the `icon` prop below, and the prefix prop can be removed.